### DISCUSSION TITLE
Simplifying theme palettes

### DISCUSSION THREAD

**Discussion by nevillepark** [Docile-Alligator/Infinity-For-Reddit#924] [Ideas]
At the moment there are [a *lot* of colour options to set](https://github.com/Docile-Alligator/Infinity-For-Reddit/blob/master/app/src/main/java/ml/docilealligator/infinityforreddit/customtheme/CustomTheme.java) (78, if I'm reading the code right!), and it makes customizing themes pretty onerous. A lot of the options are the same—e. g. in the Indigo theme, primary text, secondary text, post title, post content, and comment colour are all black/dark grey; a lot of button text is white; etc. 

Perhaps it could be condensed down to about 10 colours users can choose, by 

1) applying one colour to several different elements;  
2) automatically making text light or dark based on the value of the background colour; and
3) generating colour spectrums for things like comment vertical colour bars, post tags/flair, etc.

**Comment by Kuuchuu** [comment]
I personally like the current options. You can tweak everything to exactly how you want it.
